shared/actions/login.js

```diff
@@ -577,16 +559,19 @@ function* startLoginSaga() {
   }
 }
 
-function* reloginSaga({payload: {usernameOrEmail, passphrase}}: LoginGen.ReloginPayload) {
-  yield Saga.put(LoginGen.createSetRevokedSelf({revoked: ''}))
-  yield Saga.put(LoginGen.createSetDeletedSelf({deletedUsername: ''}))
-
-  yield Saga.call(initalizeMyCodeStateForLogin)
-  yield Saga.call(loginFlowSaga, usernameOrEmail, passphrase)
+function _relogin({payload: {usernameOrEmail, passphrase}}: LoginGen.ReloginPayload) {
+  return Saga.all([
+    Saga.put(LoginGen.createSetRevokedSelf({revoked: ''})),
+    Saga.put(LoginGen.createSetDeletedSelf({deletedUsername: ''})),
+    Saga.call(initalizeMyCodeStateForLogin),
+    Saga.call(loginFlowSaga, usernameOrEmail, passphrase),
```

Comment:
Is this semantically the same? the old code initialized first, then called loginFlowSaga.

Maybe we should add another util `Saga.sequentially` easy way to implement it could be:

```
// In util/saga
function *sequentially(effects: Array<Effects>) {
  let results = []
  for (let i = 0; i < effects.length; i++) {
    results.push(yield effects[i])
  }
  return results
}
```
Usage:
```
Saga.call(Saga.sequentially, [Saga.call(initalizeMyCodeStateForLogin), Saga.call(loginFlowSaga, usernameOrEmail, passphrase)])
```

Reply:
sounds good. my hope is ultimately to have less of this kind of logic anyways, more chaining eg loginFlowSaga would take on something which the initialize emits

Reply:
updated and actually used it twice (the outer 'all') since the old code really did this all sequentially)